core/drivers/stm32_bsec.c

```diff
@@ -625,22 +673,37 @@ TEE_Result stm32_bsec_find_otp_in_nvmem_layout(const char *name,
 		if (!nvmem_layout[i].name || strcmp(name, nvmem_layout[i].name))
 			continue;
 
-		if (otp_id)
-			*otp_id = nvmem_layout[i].otp_id;
+		return stm32_bsec_otp_setting(i, otp_id, otp_bit_offset,
+					      otp_bit_len);
+	}
+
+	DMSG("nvmem %s failed", name);
+
+	return TEE_ERROR_ITEM_NOT_FOUND;
+}
 
-		if (otp_bit_len)
-			*otp_bit_len = nvmem_layout[i].bit_len;
+TEE_Result stm32_bsec_find_otp_by_phandle(const uint32_t phandle,
+					  uint32_t *otp_id,
+					  uint8_t *otp_bit_offset,
+					  size_t *otp_bit_len)
+{
+	size_t i = 0;
 
-		DMSG("nvmem %s = %zu: %"PRId32" %zu", name, i,
-		     nvmem_layout[i].otp_id, nvmem_layout[i].bit_len);
+	if (!phandle)
+		return TEE_ERROR_GENERIC;
 
-		return TEE_SUCCESS;
+	for (i = 0; i < nvmem_layout_count; i++) {
+		if (nvmem_layout[i].phandle != phandle)
+			continue;
+
+		return stm32_bsec_otp_setting(i, otp_id, otp_bit_offset,
+					      otp_bit_len);
 	}
 
-	DMSG("nvmem %s failed", name);
+	DMSG("nvmem %u not found", phandle);
```

Comment:
Maybe an EMSG could be use here instead.

Reply:
I think and "error" trace EMSG shoud be done by caller if it is needed.... it is note needed in BSEC driver himself